frontend/packages/ceph-storage-plugin/src/components/modals/add-capacity-modal/add-capacity-modal.tsx

```diff
@@ -10,74 +9,95 @@ import {
 } from '@console/internal/components/factory';
 import { usePrometheusPoll } from '@console/internal/components/graphs/prometheus-poll-hook';
 import { k8sPatch, StorageClassResourceKind } from '@console/internal/module/k8s';
-import { PrometheusEndpoint } from '@console/internal/components/graphs/helpers';
 import { getName } from '@console/shared';
 import { OCSServiceModel } from '../../../models';
 import { OSD_CAPACITY_SIZES } from '../../../utils/osd-size-dropdown';
-import { CEPH_STORAGE_NAMESPACE, NO_PROVISIONER } from '../../../constants';
-import { labelTooltip, storageClassTooltip } from '../../../constants/ocs-install';
-import { CAPACITY_USAGE_QUERIES, StorageDashboardQuery } from '../../../constants/queries';
+import { NO_PROVISIONER } from '../../../constants';
+import {
+  labelTooltip,
+  storageClassTooltip,
+  defaultRequestSize,
+} from '../../../constants/ocs-install';
 import { OCSStorageClassDropdown } from '../storage-class-dropdown';
 import { PVsAvailableCapacity } from '../../ocs-install/pvs-available-capacity';
 import './_add-capacity-modal.scss';
+import { createDeviceSet, DeviceSet } from '../../ocs-install/ocs-request-data';
+import { getStorageSize } from '@console/kubevirt-plugin/src/selectors/selectors';
+import { cephCapacityResource } from '../../../constants/resources';
 
 const getProvisionedCapacity = (value: number) => (value % 1 ? (value * 3).toFixed(2) : value * 3);
 
+const getCurrentDeviceSet = (deviceSets: DeviceSet[], selectedSCName: string): number =>
+  deviceSets.findIndex((ds) => ds.dataPVCTemplate.spec.storageClassName === selectedSCName);
+
 export const AddCapacityModal = (props: AddCapacityModalProps) => {
   const { ocsConfig, close, cancel } = props;
+  const deviceSets: DeviceSet[] = ocsConfig?.spec?.storageDeviceSets || [];
 
-  const [response, loadError, loading] = usePrometheusPoll({
-    endpoint: PrometheusEndpoint.QUERY,
-    namespace: CEPH_STORAGE_NAMESPACE,
-    query: CAPACITY_USAGE_QUERIES[StorageDashboardQuery.CEPH_CAPACITY_USED],
-  });
+  const [response, loadError, loading] = usePrometheusPoll(cephCapacityResource);
   const [storageClass, setStorageClass] = React.useState<StorageClassResourceKind>(null);
+  /* TBD(Afreen): Show installation storage class as preselected 
+                  Change state metadata
+  */
```

Comment:
As part of enhancements and refactors in a followup PR

Reply:
It will throw an error , we are handling that case.
This TODO is to show storage class used  @ installation as the preselected one.